inst/private/cygpath.m

```diff
@@ -0,0 +1,50 @@
+%% Copyright (C) 2022 Alex Vong
+%%
+%% This file is part of OctSymPy.
+%%
+%% OctSymPy is free software; you can redistribute it and/or modify
+%% it under the terms of the GNU General Public License as published
+%% by the Free Software Foundation; either version 3 of the License,
+%% or (at your option) any later version.
+%%
+%% This software is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty
+%% of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
+%% the GNU General Public License for more details.
+%%
+%% You should have received a copy of the GNU General Public
+%% License along with this software; see the file COPYING.
+%% If not, see <https://www.gnu.org/licenses/>.
+
+%% -*- texinfo -*-
+%% @defun  cygpath ()
+%% Convert Windows native path to Cygwin POSIX-style path.
+%%
+%% @seealso{python_env_is_cygwin_like}
+%% @end defun
+
+function posix_path = cygpath (native_path)
+  args = {'-u', native_path};
+  [fin, fout, pid] = popen2 ('cygpath', args);
+  if pid == -1
+    error ('cygpath: failed to create cygpath subprocess');
+  end
+
+  assert (fclose (fin) == 0);
+
+  err = errno ('EAGAIN');
+  while err == errno ('EAGAIN')
+    pause (0.1);
+    posix_path = fgetl (fout);
+    err = errno ();
+  end
+
+  assert (ischar (posix_path));
+  assert (feof (fout));
+  assert (fclose (fout) == 0);
+
+  [retpid, status, msg] = waitpid (pid);
+  if retpid < 0
+    error ('cygpath: failed to wait for python: %s', msg);
```

Comment:
The error message doesn't look correct. Afaict, this isn't waiting for Python.

Reply:
How about lengthening it to "...wait for python to terminate"?

Reply:
Why "python"? Isn't this waiting for "cygpath"?

Reply:
Aha, I see. I probably forgot to change it.